### ISSUE TITLE
Scalar indexing when using GPU

### ISSUE THREAD

**Issue by Moelf** [Evovest/EvoTrees.jl#277]
```julia
Warning: Performing scalar indexing on task Task (runnable, started) @0x00007fe8fe4968b0.
│ Invocation of getindex resulted in scalar indexing of a GPU array.
│ This is typically caused by calling an iterating implementation of a method.
│ Such implementations *do not* execute on the GPU, but very slowly on the CPU,
│ and therefore should be avoided.
│ 
│ If you want to allow scalar iteration, use `allowscalar` or `@allowscalar`
│ to enable scalar iteration globally or for the operations in question.
└ @ GPUArraysCore ~/.julia/packages/GPUArraysCore/aNaXo/src/GPUArraysCore.jl:145
```

The training setup looks something like:
```julia
evotree_config = EvoTreeClassifier(;
        nrounds=num_round,
        eta=η,
        max_depth,
        min_weight=3.0,
        nbins=128,
    )

fit_evotree(evotree_config, dtrain;
            target_name="proc",
            fnames=string.(input_names),
            w_name="wgts",
            deval,
            metric=:logloss,
            print_every_n=5,
            verbosity=1,
            early_stopping_rounds, device="gpu")
```

**Comment by Moelf** [comment]
>metric=:logloss,

this is the problem, for classification, it needs to be `:mlogloss` -- perhaps this should throw an error?

**Comment by jeremiedb** [comment]
Yes I agree that it should have thrown an error. 
Upcomnig release for v0.17 should fix this by having both the loss and metric baked right into in `EvoTreeClassifier`: https://github.com/Evovest/EvoTrees.jl/blob/835d8a47d33a1f52b942658c6e7d8c73fec90930/src/learners.jl#L238-L239
I'm hoping to have it release by next week.
